open3d_slam/open3d_slam/src/Voxel.cpp

```diff
@@ -113,6 +89,36 @@ PointCloud VoxelizedPointCloud::toPointCloud() const {
   return ret;
 }
 
+void VoxelizedPointCloud::transform(const Transform& T) {
+  std::unordered_map<Eigen::Vector3i, AggregatedVoxel, EigenVec3iHash> voxels;
+  if (empty()) {
+    return;
+  }
+  voxels.reserve(voxels_.size());
+  for (const auto& v : voxels_) {
+    if (v.second.numAggregatedPoints_ > 0) {
+      AggregatedVoxel vTransformed(v.second);
+      vTransformed.aggregatedNormal_ = T * vTransformed.aggregatedNormal_;
+      vTransformed.aggregatedPosition_ = T * vTransformed.aggregatedPosition_;
+      voxels[v.first] = vTransformed;
+    }
+  }
+  voxels_ = std::move(voxels);
+}
+
+bool VoxelizedPointCloud::hasColors() const {
+  return isHasColors_;
+}
+
+bool VoxelizedPointCloud::hasNormals() const {
+  return isHasNormals_;
+}
+
+// Treat black as no color
+bool VoxelizedPointCloud::isValidColor(const Eigen::Vector3d& c) {
+  return (c.array().all() > 0.0) && (c.array().all() <= 1.0);
```

Comment:
what happens if something actually is black in color? Then you would just have a hole in your map, no?